rootfs/etc/nginx/lua/test/configuration_test.lua

```diff
@@ -0,0 +1,114 @@
+_G._TEST = true
+local cjson = require("cjson")
+
+function get_mocked_ngx_env()
+    local _ngx = {}
+    setmetatable(_ngx, {__index = _G.ngx})
+
+    _ngx.status = 100
+    _ngx.var = {}
+    _ngx.req = {
+        read_body = function() end,
+        get_body_file = function() end,
+    }
+    return _ngx
+end
+
+local unmocked_ngx = _G.ngx
+local configuration = require("configuration")
+
+describe("Configuration", function()
+    before_each(function()
+        _G.ngx = get_mocked_ngx_env()
+    end)
+
+    after_each(function()
+        _G.ngx = unmocked_ngx
+    end)
+
+    describe("handle_servers()", function()
+        it("should not accept non POST methods", function()
+            ngx.var.request_method = "GET"
+            
+            local s = spy.on(ngx, "print")
+            assert.has_no.errors(configuration.handle_servers)
+            assert.spy(s).was_called_with("Only POST requests are allowed!")
+            assert.same(ngx.status, ngx.HTTP_BAD_REQUEST)
+        end)
+
+        it("should ignore servers that don't have hostname or pemCertKey set", function()
+            ngx.var.request_method = "POST"
+            local mock_servers = cjson.encode({
+                {
+                    hostname = "hostname",
+                    sslCert = {}
+                },
+                {
+                    sslCert = {
+                        pemCertKey = "pemCertKey"
+                    }
+                }
+            })
+            ngx.req.get_body_data = function() return mock_servers end
+
+            local s = spy.on(ngx, "log")
+            assert.has_no.errors(configuration.handle_servers)
+            assert.spy(s).was_called_with(ngx.WARN, "hostname or pemCertKey are not present")
+            assert.same(ngx.status, ngx.HTTP_CREATED)
+        end)
+
+        it("should successfully update certificates and keys for each host", function()
+            ngx.var.request_method = "POST"
+            local mock_servers = cjson.encode({
+                {
+                    hostname = "hostname",
+                    sslCert = {
+                        pemCertKey = "pemCertKey"
+                    }
+                }
+            })
+            ngx.req.get_body_data = function() return mock_servers end
+
+            assert.has_no.errors(configuration.handle_servers)
+            assert.same(ngx.status, ngx.HTTP_CREATED)
```

Comment:
you should also assert that the data is stored in the shared dictionary - that's the most important outcome of calling `handle_servers`